components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -1941,6 +1991,32 @@ private ServiceProvider unmarshalSP(String spTemplateXml, String tenantDomain)
         }
     }
 
+    /**
+     * Convert yml file of service provider to object.
+     *
+     * @param spFileContent xml string of the SP and file name.
+     * @param tenantDomain  tenant domain name.
+     * @return Service Provider.
+     * @throws IdentityApplicationManagementException Identity Application Management Exception.
+     */
+    private ServiceProvider deserializationSP(SpFileContent spFileContent, String tenantDomain)
+            throws IdentityApplicationManagementException {
+
+        if (StringUtils.isEmpty(spFileContent.getContent())) {
+            throw new IdentityApplicationManagementException(String.format("Empty Service Provider configuration file" +
+                    " %s uploaded by tenant: %s", spFileContent.getFileName(), tenantDomain));
+        }
+
+        try {
+            Yaml yaml = new Yaml(new Constructor(ServiceProvider.class));
+            ServiceProvider serviceProvider = yaml.loadAs(spFileContent.getContent(), ServiceProvider.class);
+            return serviceProvider;
+        } catch (Exception e) {
```

Comment:
Let's not Catch the Exception class here since it might shadow all the exceptions. Try to catch only the specific exception.

Reply:
Changed it to catch only a YAMLException